### PR TITLE
[FIX] l10n_cl_invoicing_policy

### PR DESCRIPTION
**Pull request by max3903**
Allow to select "Invoice" and "Electronic Guide" for employees and addresses

### CODE REVIEW

**Review by nelsonramirezs — APPROVED** [review]
LGTM!

File: l10n_cl_invoicing_policy/models/res_partner.py [hunk 1]
```diff
@@ -23,19 +23,21 @@ def _default_invoicing_policy(self):
...
+        * Electronic Guide: Only for companies and their children.
+          1 invoice at the end of the month for all the delivery orders of
+          that month.""")
 
-    @api.constrains('is_company', 'invoicing_policy')
+    @api.constrains('is_company', 'invoicing_policy', 'parent_id')
     def check_invoicing_policy(self):
-        if self.invoicing_policy == 'ticket':
-            if self.is_company:
+        if self.is_company or self.parent_id:
```
Comment (dreispt) [inline l10n_cl_invoicing_policy/models/res_partner.py]: This is incorrect: it doesn't allow for child Addresses of individuals to be set to `ticket`.
See #97 for an implementation suggestion.

**Review by dreispt — CHANGES_REQUESTED** [review]

File: l10n_cl_invoicing_policy/models/res_partner.py (hunk 1 above)
Comment (max3903) [reply l10n_cl_invoicing_policy/models/res_partner.py]: @dreispt Yes, this is the desired behavior... Right @nelsonramirezs?

File: l10n_cl_invoicing_policy/models/res_partner.py [hunk 2]
```diff
@@ -23,21 +26,24 @@ def _default_invoicing_policy(self):
...
-                                  ' to companies.'))
+        rec = self
+        while rec.parent_id:
+            rec = rec.parent_id
+        if rec.is_company and self.invoicing_policy == 'ticket':
+            raise UserError(_(
+                'The invoicing policy Ticket only applies to individuals.'))
+        if not rec.is_company and self.invoicing_policy != 'ticket':
+            raise UserError(_(
+                'The selected invoicing policy only applies to companies.'))
```
Comment (dreispt) [inline l10n_cl_invoicing_policy/models/res_partner.py]: You are rechecking the parent invoicing policy every time when adding/editing a child address.
Seems redundant to me.
Why not just check on parent edits?

**Review by dreispt — APPROVED** [review]

**Review by max3903 — COMMENTED** [review]
/ocabot merge patch

**Pull request merged**